src/cpp/fastdds/xtypes/type_representation/TypeObjectUtils.cpp

```diff
@@ -2050,6 +2050,175 @@ void TypeObjectUtils::plain_collection_type_identifier_header_consistency(
     }
 }
 
+EquivalenceKind TypeObjectUtils::get_map_component_equiv_kind_for_consistency(
+        const TypeIdentifier& identifier,
+        bool is_key)
+{
+    if (is_direct_hash_type_identifier(identifier))
+    {
+        TypeObject type_object;
+        if (eprosima::fastdds::dds::RETCODE_OK ==
+                type_object_registry_observer().get_type_object(identifier, type_object))
+        {
+            return type_object._d();
+        }
+        else
+        {
+            throw InvalidArgumentError("Given TypeIdentifier is not found in TypeObjectRegistry");
+        }
+    }
+    else if (is_indirect_hash_type_identifier(identifier))
+    {
+        if (is_key)
+        {
+            // indirect hash for key is not allowed
+            return TK_NONE;
+        }
+        EquivalenceKind element_equiv_kind;
+        EquivalenceKind key_equiv_kind;
+
+        switch (identifier._d())
+        {
+            case TI_PLAIN_SEQUENCE_SMALL:
+                return get_map_component_equiv_kind_for_consistency(*identifier.seq_sdefn().element_identifier(),
+                               false);
+            case TI_PLAIN_SEQUENCE_LARGE:
+                return get_map_component_equiv_kind_for_consistency(*identifier.seq_ldefn().element_identifier(),
+                               false);
+            case TI_PLAIN_ARRAY_SMALL:
+                return get_map_component_equiv_kind_for_consistency(
+                    *identifier.array_sdefn().element_identifier(), false);
+            case TI_PLAIN_ARRAY_LARGE:
+                return get_map_component_equiv_kind_for_consistency(
+                    *identifier.array_ldefn().element_identifier(), false);
+            case TI_PLAIN_MAP_SMALL:
+                if (is_fully_descriptive_type_identifier(*identifier.map_sdefn().element_identifier()) &&
+                        is_fully_descriptive_type_identifier(*identifier.map_sdefn().key_identifier()))
+                {
+                    return EK_BOTH;
+                }
+
+                element_equiv_kind = get_map_component_equiv_kind_for_consistency(
+                    *identifier.map_sdefn().element_identifier(), false);
+                key_equiv_kind = get_map_component_equiv_kind_for_consistency(
+                    *identifier.map_sdefn().key_identifier(), false);
```

Comment:
```suggestion
                    *identifier.map_sdefn().key_identifier(), true);
```

Reply:
removed in [a989bd3](https://github.com/eProsima/Fast-DDS/pull/4496/commits/a989bd32df198abf53da0faff2c924f05a240c47)